oak_tensorflow_service/src/tflite.rs

```diff
@@ -80,18 +80,27 @@ impl TfliteModel {
         let input_bytes_len = input_bytes
             .len()
             .try_into()
-            .map_err(|error| anyhow!("Failed to convert usize to u64: {}", error))?;
-        // TODO(#3297): Allocate enough bytes for the TensorFlow Lite model.
-        let mut output_bytes = vec![0; input_bytes.len()];
-        let mut output_bytes_len = 0u64;
+            .map_err(|error| anyhow!("Failed to convert input bytes length to u64: {}", error))?;
+
+        // TensorFlow Lite is the owner of the corresponding `output_bytes` buffer.
+        let output_bytes_ptr: *mut *const u8 = &mut core::ptr::null();
+        let mut output_bytes_len_ptr = 0u64;
+
         let _ = unsafe {
             tflite_run(
                 input_bytes.as_ptr(),
                 input_bytes_len,
-                output_bytes.as_mut_ptr(),
-                &mut output_bytes_len as *mut u64,
+                output_bytes_ptr,
+                &mut output_bytes_len_ptr as *mut u64,
             )
         };
-        Ok(output_bytes)
+
+        let output_bytes_len = output_bytes_len_ptr
+            .try_into()
+            .map_err(|error| anyhow!("Failed to get output bytes length from bytes: {}", error))?;
+
+        let output_bytes =
+            unsafe { alloc::slice::from_raw_parts(*output_bytes_ptr, output_bytes_len) };
```

Comment:
What happens to the slice after this? AFAICT the vector below is a new allocation in which the bytes are copied, but what about the existing pointer?

Reply:
Current idea was to point this pointer to the Tensor Arena output in C++ (which is always the same buffer). Because without `malloc` TFLM cannot allocate new buffers at all.

But probably it's safer to allocate everything on the Rust side.